content/tokio/tutorial/bridging.md

```diff
@@ -0,0 +1,384 @@
+---
+title: "Bridging with sync code"
+---
+
+In the examples we have seen so far, we marked the main function with
+`#[tokio::main]` and made the entire project asynchronous. However, this is not
+desirable for all projects. For instance, a GUI application might want to run
+the GUI code on the main thread and run a Tokio runtime next to it on another
+thread.
+
+This page explains how you can isolate async/await to a small part of your
+project.
+
+# What `#[tokio::main]` expands to
+
+The `#[tokio::main]` macro is a macro that replaces your main function with a
+non-async main function that starts a runtime and then calls your code. For
+instance, this:
+```rust
+#[tokio::main]
+async fn main() {
+    println!("Hello world");
+}
+```
+is turned into this:
+```
+fn main() {
+    tokio::runtime::Builder::new_multi_thread()
+        .enable_all()
+        .build()
+        .unwrap()
+        .block_on(async {
+            println!("Hello world");
+        })
+}
+```
+by the macro. To use async/await in our own projects, we can do something
+similar where we leverage the [`block_on`] method to enter the asynchronous
+context where appropriate.
+
+# A synchronous interface to mini-redis
+
+In this section, we will go through how to build a synchronous interface to
+mini-redis by storing a `Runtime` object and using its `block_on` method.
+In the following sections, we will discuss some alternate approaches and when
+you should use each approach.
+
+The interface that we will be wrapping is the asynchronous [`Client`] type. It
+has several methods, and we will implement a blocking version of the following
+methods:
+
+ * [`Client::get`]
+ * [`Client::set`]
+ * [`Client::set_expires`]
+ * [`Client::publish`]
+ * [`Client::subscribe`]
+
+To do this, we introduce a new file called `src/blocking_client.rs` and
+initialize it with a wrapper struct around the async `Client` type:
+```rs
+use tokio::net::ToSocketAddrs;
+use tokio::runtime::Runtime;
+
+pub use crate::client::Message;
+
+/// Established connection with a Redis server.
+pub struct Client {
+    /// The asynchronous `Client`.
+    inner: crate::client::Client,
+
+    /// A `current_thread` runtime for executing operations on the
+    /// asynchronous client in a blocking manner.
+    rt: Runtime,
+}
+
+pub fn connect<T: ToSocketAddrs>(addr: T) -> crate::Result<Client> {
+    let rt = tokio::runtime::Builder::new_current_thread()
+        .enable_all()
+        .build()?;
+
+    // Call the asynchronous connect method using the runtime.
+    let inner = rt.block_on(crate::client::connect(addr))?;
+
+    Ok(Client { inner, rt })
+}
+```
+Here, we have included the constructor function as our first example of how to
+execute asynchronous methods in a non-async context. We do this using the
+[`block_on`] method on the Tokio [`Runtime`] type, which executes an
+asynchronous method and returns its result.
+
+One important detail is the use of the [`current_thread`] runtime. Normally when
```

Comment:
```suggestion
One important detail is the use of the [`current_thread`] runtime. Usually when
```

I would pick a different word than "Normally" as it implies using the current_thread runtime is ab-normal. "Often" could work as well.